pkg/graph/email.resolvers.go

```diff
@@ -332,6 +333,191 @@ func (r *mutationResolver) DeleteDomain(ctx context.Context, input model.DeleteD
+	currentUser, err := middleware.GetUser(ctx)
+	if err != nil {
+		return nil, errors.New("no user found in the context")
+	}
+	if !currentUser.HasPermissionAdmin() {
+		return nil, errors.New("no permission")
+	}
+
+	group := model.Group{
+		Name:        input.Name,
+		Description: input.Description,
+		Domain:      input.Domain,
+		Users:       input.Users,
+	}
+
+	generatedName := input.Name + "@" + input.Domain
+	// Insert new email account of type group
+	account := db.EmailAccount{
+		Name:           generatedName,
+		Type:           "GROUP",
+		Description:    *input.Description,
+		OrganisationID: currentUser.OrganisationID,
+	}
+	err = r.DB.NewInsert().Model(&account).Scan(ctx)
+
+	group.ID = account.ID
+
+	// For each user in the group, create a new email group member
+	for _, user := range input.Users {
+		groupMember := db.EmailGroupMember{
+			Name:           *user,
+			OrganisationID: currentUser.OrganisationID,
+			MemberOf:       generatedName,
+		}
+		err = r.DB.NewInsert().Model(&groupMember).Scan(ctx)
+	}
+
+	return &group, nil
+}
+
+// UpdateGroup is the resolver for the updateGroup field.
+func (r *mutationResolver) UpdateGroup(ctx context.Context, input model.UpdateGroupInput) (*model.Group, error) {
+	currentUser, err := middleware.GetUser(ctx)
+	if err != nil {
+		return nil, errors.New("no user found in the context")
+	}
+	if !currentUser.HasPermissionAdmin() {
+		return nil, errors.New("no permission")
+	}
+	// Fetch the current group
+	var group db.EmailAccount
+	err = r.DB.NewSelect().Model(&group).Where("id = ?", input.ID).Where("organisation_id = ?", currentUser.OrganisationID).Scan(ctx)
+
+	// Update email account of type group
+	account := db.EmailAccount{
+		ID:             input.ID,
+		Name:           *input.Name + "@" + *input.Domain,
+		Type:           "GROUP",
+		Description:    *input.Description,
+		OrganisationID: currentUser.OrganisationID,
+		CreatedAt:      group.CreatedAt,
+	}
+
+	// Check if the group name has changed
+	if account.Name != group.Name {
+		// Update group members with new group name
+		var groupMembers []db.EmailGroupMember
+		err = r.DB.NewSelect().Model(&groupMembers).Where("member_of = ?", group.Name).Scan(ctx)
+
+		// Update each entry
+		for _, member := range groupMembers {
+			member.MemberOf = account.Name
+			err = r.DB.NewUpdate().Model(&member).Scan(ctx)
+		}
+	}
+
+	// Update the group
+	_, err = r.DB.NewUpdate().Model(&account).Where("id = ?", input.ID).Returning("*").Exec(ctx)
+
+	// Get the current group members
+	var groupMembers []db.EmailGroupMember
+	err = r.DB.NewSelect().Model(&groupMembers).Where("member_of = ?", account.Name).Scan(ctx)
+
+	// Remove the ones not in input.Users
+	for _, member := range groupMembers {
+		found := false
+		for _, user := range input.Users {
+			if *user == member.Name {
+				found = true
+			}
+		}
+
+		if !found {
+			// Delete the group member
+			_, err = r.DB.NewDelete().Model(&member).Where("id = ?", member.ID).Exec(ctx)
```

Comment:
Is this a hard delete? I am not sure. If we are using a soft delete, we need to update the Stalwart sql queries to exclude deleted rows.